engine/runtime-instrument-common/src/main/java/org/enso/interpreter/instrument/Observable.java

```diff
@@ -0,0 +1,115 @@
+package org.enso.interpreter.instrument;
+
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import java.util.UUID;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import org.enso.interpreter.service.GuestExecutionService;
+
+public class Observable {
```

Comment:
- could this generic(?) observable support be moved out of `runtime-instrument-common` into own project?
- it would be easier to analyze the observables system in isolation:
   - easier to unit test
   - easier to built a mental model of the observables framework
   - easier to document it
- ideally such a project would have no other dependencies than JVM
- thus the conceptual surface would be way smaller than of the whole `runtime-instrument-common` project